monkey/infection_monkey/utils/commands.py

```diff
@@ -0,0 +1,61 @@
+import logging
+
+from infection_monkey.model import CMD_CARRY_OUT, CMD_EXE, MONKEY_ARG
+
+LOG = logging.getLogger(__name__)
+
+
+def build_monkey_commandline(target_host, depth, vulnerable_port, location=None):
+    from infection_monkey.config import GUID
+
+    return "".join(
+        build_monkey_commandline_explicitly(
+            GUID,
+            target_host.default_tunnel,
+            target_host.default_server,
+            depth,
+            location,
+            vulnerable_port,
+        )
+    )
+
+
+def build_monkey_commandline_explicitly(
```

Comment:
Notice how all usages of this method are followed by `get_monkey_commandline_windows` or `get_monkey_commandline_linux` which indicates a lack of abstraction and encapsulation. Out of scope probably, but we could open up a PR after this if you think this is annoying.

Reply:
`build_monkey_commandline()` also uses it. The exploiters use `build_monkey_commandline()`. It's definitely annoying and there will be a followup PR.